core/dbt/parser/schemas.py

```diff
@@ -132,11 +125,10 @@ def add(
             quote = None
         self.column_info[column.name] = ColumnInfo(
             name=column.name,
-            description=description,
-            data_type=data_type,
-            constraints=constraints,
-            constraints_check=constraints_check,
-            meta=meta,
+            description=column.description,
+            data_type=column.data_type,
+            constraints=[ColumnLevelConstraint.from_dict(c) for c in column.constraints],
```

Comment:
This will blow up if the "type" in the constraint isn't valid. You should make a test case with a column level constraint with an invalid type string.

Reply:
@gshank I verified that the line you highlighted throws a mashumaro.exceptions.InvalidFieldValue exception. Should we catch it here and raise a ParsingError exception instead, or do you think we should be validating this somewhere else?

Reply:
I validated separately in a similar circumstance. In the "patch" method on ParsedNode in nodes.py, I did: 
```
               if AccessType.is_valid(patch.access):
                    self.access = AccessType(patch.access)
                else:
                    raise InvalidAccessTypeError(
                        unique_id=self.unique_id,
                        field_value=patch.access,
                    )

```
Added an "is_valid" method to the AccessType enum.

Reply:
I would validate right there before you want to construct those ColumnLevelConstraints.

Reply:
Not sure whether we want to proliferate these kind of errors or not. I suppose we could do something more generic...